src/services/profile/followProfile.ts

```diff
@@ -1,31 +1,48 @@
+import type { ComputedRef } from 'vue'
+import { redirect } from '../../router'
+
+import type { AuthorizationError } from '../../types/error'
+
 import { request } from '../index'
+
 import createAsyncProcess from '../../utils/create-async-process'
-import { ComputedRef } from 'vue'
+import { mapAuthorizationResponse } from '../../utils/map-checkable-response'
+import { Either, fail, success } from '../../utils/either'
 
-export async function postFollowProfile (username: string) {
-  return request.post<ProfileResponse>(`/profiles/${username}/follow`).then(res => res.profile)
+export async function postFollowProfile (username: string): Promise<Either<AuthorizationError, Profile>> {
+  const result1 = await request.checkablePost<ProfileResponse>(`/profiles/${username}/follow`)
+  const result2 = mapAuthorizationResponse<ProfileResponse>(result1)
+
+  if (result2.isOk()) return success(result2.value.profile)
+  return fail(result2.value)
 }
 
-export async function deleteFollowProfile (username: string) {
-  return request.delete<ProfileResponse>(`/profiles/${username}/follow`).then(res => res.profile)
+export async function deleteFollowProfile (username: string): Promise<Either<AuthorizationError, Profile>> {
+  const result1 = await request.checkableDelete<ProfileResponse>(`/profiles/${username}/follow`)
+  const result2 = mapAuthorizationResponse<ProfileResponse>(result1)
+
+  if (result2.isOk()) return success(result2.value.profile)
+  return fail(result2.value)
 }
 
 interface UseFollowProps {
   username: ComputedRef<string>
   following: ComputedRef<boolean>
+  onUpdate: (profile: Profile) => void
 }
 
-export function useFollow ({ username, following }: UseFollowProps) {
-  async function toggleFollow (): Promise<Profile> {
-    let profile = null
+export function useFollow ({ username, following, onUpdate }: UseFollowProps) {
+  async function toggleFollow () {
+    let response: Either<AuthorizationError, Profile>
 
     if (following.value === true) {
-      profile = await deleteFollowProfile(username.value)
+      response = await deleteFollowProfile(username.value)
     } else {
-      profile = await postFollowProfile(username.value)
+      response = await postFollowProfile(username.value)
     }
 
-    return profile
+    if (response.isOk()) onUpdate(response.value)
+    else redirect('login')
```

Comment:
Perhaps we can `routerPush()` instead of `redirect()` to avoid confusion with `router.replace`?

Reply:
Yeap, ok
I will replace it, every where in codebase, soon